File: examples/ruby/unsetNotes.rb
#!/usr/bin/env ruby
#
## 
## @file    unsetNotes.py
## @brief   unset notes for each element
## @author  Akiya Jouraku
## 
## This file is part of libSBML.  Please visit http://sbml.org for more
## information about SBML, and the latest version of libSBML.
## 


require 'libSBML'


if (ARGV.size != 2):
    print("\nUsage: unsetNotes <input-filename> <output-filename>" + "\n" + "\n")
    exit(1);
end

filename = ARGV[0];

document = LibSBML::readSBML(filename)


errors = document.getNumErrors

if (errors > 0)
    document.printErrors
    return errors;
end

m = document.getModel
m.unsetNotes

m.getNumReactions.times do |i|
    re = m.getReaction(i)      
    re.unsetNotes

    re.getNumReactants.times do |j|
        rt = re.getReactant(j)
        rt.unsetNotes          
    end
    re.getNumProducts.times do |j|
        rt = re.getProduct(j)
        rt.unsetNotes
    end
    re.getNumModifiers.times do |j|
        md = re.getModifier(j)
        md.unsetNotes
    end
    if (re.isSetKineticLaw)
        kl = re.getKineticLaw
        kl.unsetNotes

        kl.getNumParameters.times do |j|
            pa = kl.getParameter(j)
            pa.unsetNotes
        end
    end
end
m.getNumSpecies.times do |i|
    sp = m.getSpecies(i)
    sp.unsetNotes
end
m.getNumCompartments.times do |i|
    sp = m.getCompartment(i)
    sp.unsetNotes
end
m.getNumFunctionDefinitions.times do |i|
    sp = m.getFunctionDefinition(i)
    sp.unsetNotes
end
m.getNumUnitDefinitions.times do |i|
    sp = m.getUnitDefinition(i)
    sp.unsetNotes
end
m.getNumParameters.times do |i|
    sp = m.getParameter(i)
    sp.unsetNotes
end
m.getNumRules.times do |i|
    sp = m.getRule(i)
    sp.unsetNotes
end
m.getNumInitialAssignments.times do |i|
    sp = m.getInitialAssignment(i)
    sp.unsetNotes
end
m.getNumEvents.times do |i|
    sp = m.getEvent(i)
    sp.unsetNotes

    sp.getNumEventAssignments.times do |j|
        ea = sp.getEventAssignment(j)
        ea.unsetNotes
    end
end
m.getNumSpeciesTypes.times do |i|
    sp = m.getSpeciesType(i)
    sp.unsetNotes
end
m.getNumConstraints.times do |i|
    sp = m.getConstraint(i)
    sp.unsetNotes
end
LibSBML::writeSBML(document, ARGV[1])

exit(errors);

File: examples/perl/unsetNotes.pl
#!/usr/bin/env perl
# -*-Perl-*-
## 
## @file    unsetNotes.pl
## @brief   unset notes for each element
## @author  Akiya Jouraku
## 
## This file is part of libSBML.  Please visit http://sbml.org for more
## information about SBML, and the latest version of libSBML.
## 

use LibSBML;
no strict;

if ($#ARGV != 1 ) {
	print "Usage: unsetNotes <input-filename> <output-filename>\n";
	exit 1;
}

$filename = $ARGV[0];

$document = LibSBML::readSBML($filename);

$errors = $document->getNumErrors();

if ($errors > 0) {
    $document->printErrors();
    exit $errors;
}

$m = $document->getModel();
$m->unsetNotes();

for ($i = 0; $i < $m->getNumReactions(); $i++) {
    $re = $m->getReaction($i);
    $re->unsetNotes();

    for ($j = 0; $j < $re->getNumReactants(); $j++) {
        $rt = $re->getReactant($j);
        $rt->unsetNotes();
    }  
    for ($j = 0; $j < $re->getNumProducts(); $j++) {
        $rt = $re->getProduct($j);
        $rt->unsetNotes();
    }
    for ($j =0; $j < $re->getNumModifiers(); $j++) {
        $md = $re->getModifier($j);
        $md->unsetNotes();
    }
    if ($re->isSetKineticLaw()) {
        $kl = $re->getKineticLaw();
        $kl->unsetNotes();

        for ($j =0; $j < $kl->getNumParameters(); $j++) {
            $pa = $kl->getParameter(j);
            $pa->unsetNotes();
        }
     }
}
for ($i = 0; $i < $m->getNumSpecies(); $i++) {
    $sp = $m->getSpecies($i);
    $sp->unsetNotes();
}
for ($i = 0; $i <$m->getNumCompartments(); $i++) {
    $sp = $m->getCompartment($i);
    $sp->unsetNotes();
}
for ($i = 0; $i <$m->getNumFunctionDefinitions(); $i++) {
    $sp = $m->getFunctionDefinition($i);
    $sp->unsetNotes();
}
for ($i = 0; $i <$m->getNumUnitDefinitions(); $i++) {
    $sp = $m->getUnitDefinition($i);
    $sp->unsetNotes();
}
for ($i = 0; $i <$m->getNumParameters(); $i++) {
    $sp = $m->getParameter($i);
    $sp->unsetNotes();
}
for ($i = 0; $i <$m->getNumRules(); $i++) {
    $sp = $m->getRule($i);
    $sp->unsetNotes();
}
for ($i = 0; $i <$m->getNumInitialAssignments(); $i++) {
    $sp = $m->getInitialAssignment($i);
    $sp->unsetNotes();
}
for ($i = 0; $i <$m->getNumEvents(); $i++) {
    $sp = $m->getEvent($i);
    $sp->unsetNotes();

    for ($j =0; $j <$sp->getNumEventAssignments(); $j++) {
        $ea = $sp->getEventAssignment($j);
        $ea->unsetNotes();
    }
}
for ($i = 0; $i <$m->getNumSpeciesTypes(); $i++) {
    $sp = $m->getSpeciesType($i);
    $sp->unsetNotes();
}
for ($i = 0; $i <$m->getNumConstraints(); $i++) {
    $sp = $m->getConstraint($i);
    $sp->unsetNotes();
}
LibSBML::writeSBML($document, $ARGV[1]);

exit $errors;

